src/compiler/checker.ts

```diff
@@ -33675,15 +33675,32 @@ namespace ts {
                 else if (isChildOfNodeWithKind(node, SyntaxKind.EnumMember)) {
                     diagnosticMessage = Diagnostics.Octal_literals_are_not_allowed_in_enums_members_initializer_Use_the_syntax_0;
                 }
-                if (diagnosticMessage) {
-                    const withMinus = isPrefixUnaryExpression(node.parent) && node.parent.operator === SyntaxKind.MinusToken;
-                    const literal = (withMinus ? "-" : "") + "0o" + node.text;
-                    return grammarErrorOnNode(withMinus ? node.parent : node, diagnosticMessage, literal);
-                }
             }
+
+            if (!diagnosticMessage && numericLiteralValueImpreciselyLarge(node.text)) {
+                diagnosticMessage = Diagnostics.Numeric_literals_with_absolute_values_equal_to_2_53_or_greater_are_too_large_to_be_represented_accurately_as_an_integer;
+            }
+
+            if (diagnosticMessage) {
+                const withMinus = isPrefixUnaryExpression(node.parent) && node.parent.operator === SyntaxKind.MinusToken;
+                const literal = (withMinus ? "-" : "") + "0o" + node.text;
+                return grammarErrorOnNode(withMinus ? node.parent : node, diagnosticMessage, literal);

```

Comment:
This is feeding in a userless `literal` parameter`. It's not a problem per se, but it is very weird to do.